pkg/daemon/update.go

```diff
@@ -423,7 +427,34 @@ func (dn *CoreOSDaemon) applyOSChanges(mcDiff machineConfigDiff, oldConfig, newC
 	}
 
 	if dn.nodeWriter != nil {
-		dn.nodeWriter.Eventf(corev1.EventTypeNormal, "OSUpdateStaged", "Changes to OS staged")
+		// We send out the event OSUpdateStaged synchronously to ensure it is recorded.
+		// Remove this when we can ensure all events are sent before exiting.
+		ref, err := reference.GetReference(scheme.Scheme, dn.node)
+		if err != nil {
+			return err
+		}
+		t := metav1.Time{Time: time.Now()}
+
+		if err = wait.PollImmediate(200*time.Millisecond, 5*time.Second, func() (done bool, err error) {
```

Comment:
how about

```go
event := // build up the event here
if _, err := dn.kubeClient.CoreV1().Events(metav1.NamespaceDefault).Create(context.TODO(), event); err != nil{
  utilruntime.HandleError(err)
}
```

This would give synchronous behavior for the writing of the event and a log message if it fails.  If you get neither, then it means you will probably want to make an update to the sig-term handling to emit this if necessary before shutdown.